lc0/src/neural/cache.cc

```diff
@@ -63,7 +63,7 @@ void CachingComputation::ComputeBlocking() {
   // Fill cache with data from NN.
   for (const auto& item : batch_) {
     if (item.idx_in_parent == -1) continue;
-    auto req = std::make_unique<CachedNNRequest>();
+    auto req = std::make_unique<CachedNNRequest>(item.probabilities_to_cache.size());
```

Comment:
That will create an array of N empty elements in req->p, and below in line 69 it will add N more entries.
It should be `req->p.reserve(N)` instead (right here, no need to do that in constructor).
Although I'd expect that preallocating of small amount (typical position has ~30 moves) of small POD values won't get any noticeable speedup.

Reply:
Doh! Yeah, what am I doing...

This object makes up the bulk of NNCache so it can't hurt for memory.